repos/system_upgrade/el7toel8/actors/copyinitramfstoboot/libraries/library.py

```diff
@@ -0,0 +1,51 @@
+from os import path
+from shutil import copyfile
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import BootContent
+
+INITRAM_FILENAME = 'initramfs-upgrade.x86_64.img'
+KERNEL_FILENAME = 'vmlinuz-upgrade.x86_64'
+
+
+def copy_initramfs_to_boot():
+    files_to_copy = get_files_to_copy()
+    copy_files(files_to_copy)
+    message_copied_files()
+
+
+def get_files_to_copy():
+    files_to_copy = {}
+    for filename in KERNEL_FILENAME, INITRAM_FILENAME:
+        files_to_copy[filename] = {'src_path': get_src_filepath(filename), 'dst_path': get_dst_filepath(filename)}
+    return files_to_copy
+
+
+def get_src_filepath(filename):
+    src_filepath = api.get_file_path(filename)
+    if src_filepath is None:
+        raise StopActorExecutionError('Could not find {0} in the following paths: {2}'
+                                      .format(filename, ' '.join(api.files_paths())),
+                                      details={'Details': 'You may want to try to reinstall'
+                                                          ' the "leapp-repository" package'})
+    return src_filepath
+
+
+def get_dst_filepath(filename):
+    return path.join('/boot', filename)
+
+
+def copy_files(files_to_copy):
+    for filename in files_to_copy.keys():
+        try:
+            copyfile(files_to_copy[filename]['src_path'], files_to_copy[filename]['dst_path'])
+        except IOError as err:
+            raise StopActorExecutionError('Could not copy {0} to /boot'.format(files_to_copy[filename]['src_path']),
+                                          details={'Details': str(err)})
```

Comment:
here the 'details' make more sense. And I would use a lower case D for the key.

Reply:
Done.